src/__tests__/utils/transform.tsx

```diff
@@ -1,51 +1,120 @@
+import * as Fs from 'fs';
 import * as ts from 'typescript';
-import * as fs from 'fs';
+import { Volume } from 'memfs';
 import * as path from 'path';
+import * as resolveModule from 'resolve';
+import pkg from '../../../package.json';
+import { copy, mkdirp } from './memfs';
 
 const printer = ts.createPrinter();
 
+declare module 'resolve' {
+  export interface SyncOpts extends resolveModule.Opts {
+    /** how to read files synchronously (defaults to fs.readFileSync) */
+    readFileSync?: (file: string, charset: string) => string | Buffer;
+    /** function to synchronously test whether a file exists */
+    isFile?: (file: string) => boolean;
+    /** function to synchronously test whether a directory exists */
+    isDirectory?: (directory: string) => boolean;
+  }
+}
+
 type ProgramTransformer = (program: ts.Program) => ts.TransformerFactory<ts.SourceFile>;
 
-/**
- * This creates a full project which will resolve all modules.
- * Only use this when wanting to test imports tbh. It's slow.
- */
-export const createFullTransform = (
-  programTransformer: ProgramTransformer,
-  dir: string
-) => (sources: {
+interface Sources {
   /**
    * This is the root file for the transform.
    */
   index: string;
   [key: string]: string;
-}): Promise<string> => {
+}
+
+/**
+ * This creates a full project which will resolve all modules.
+ * Only use this when wanting to test imports tbh. It's slow.
+ */
+export const createFullTransform = (programTransformer: ProgramTransformer) => (
+  sources: Sources
+): Promise<string> => {
   return new Promise((resolve, reject) => {
-    try {
-      fs.rmdirSync(`${dir}/.tmp`, { recursive: true });
-    } catch {}
-
-    fs.mkdirSync(`${dir}/.tmp`);
-    const files: string[] = [];
-    Object.keys(sources).forEach(key => {
-      const source = sources[key];
-      const filename = `${key}.tsx`;
-      const filepath = path.resolve(`${dir}/.tmp/${filename}`);
-      files.push(filepath);
-      fs.writeFileSync(filepath, source);
-    });
-
-    const [rootFile] = files;
+    const fs = (Volume.fromJSON(
+      Object.entries(sources).reduce<Record<string, string>>(
+        (acc, [name, content]) => ({
+          ...acc,
+          [`/${name}.tsx`]: content,
+        }),
+        {}
+      )
+    ) as unknown) as typeof Fs;
+
+    // Create mock "@untitled/css-in-js-project"
+    createMockModule(pkg.name, fs);
+
     const config: ts.CompilerOptions = {
       jsx: ts.JsxEmit.Preserve,
+      lib: ['/node_modules/typescript/lib/lib.esnext.full.d.ts'],
       module: ts.ModuleKind.ESNext,
+      moduleResolution: ts.ModuleResolutionKind.NodeJs,
       suppressImplicitAnyIndexErrors: true,
+      resolveJsonModule: true,
+      skipLibCheck: true,
```

Comment:
oh nice - would this have sped up the project creation?

Reply:
Yes, about 10% for our use case according to my tests.